docs/usage/models.md

```diff
@@ -713,29 +713,28 @@ class DataModel(BaseModel):
 
 class Response(BaseModel, Generic[DataT]):
     data: Optional[DataT] = None
-    error: Optional[Error] = Field(validate_default=True)
-
-    @field_validator('error')
-    def check_consistency(cls, v, info):
-        if v is not None and info.data['data'] is not None:
-            raise ValueError('must not provide both data and error')
-        if v is None and info.data.get('data') is None:
-            raise ValueError('must provide data or error')
-        return v
```

Comment:
This validate default stuff doesn't have anything to do with generic models (the docs section). I'm simplifying the example by removing it.

Reply:
I liked this example for the sake of showing how to build a type with a "common"/"universal" error field, and generic "data" field, as one approach to getting consistent error handling in a web API. (Essentially functioning as a field-name-tagged union.) (I vaguely recall having written this example myself, maybe I'm misremembering..)

I liked the example as I felt like it showed a bit more complex/"real world" usage. But perhaps the complexity distracts from the main point.

Either way, I think we should eventually make a point of having more "intermediate" level examples somewhere in the docs. But I guess that's something @tpdorsey will be tracking.